### ISSUE TITLE
False positive CVE excluded by dependencycheck-base-suppression.xml is still reported

### ISSUE THREAD

**Issue by itineric** [dependency-check/DependencyCheck#7094]
**Describe the bug**
The false positive that seems to be excluded by #4227 si still reported

**Version of dependency-check used**
cli 10.0.4

**To Reproduce**
Run dependency check againts simple-xml-safe.jar

**Expected behavior**
The FP should not be reported

**Additional context**
The HTML report gives the following entry:
![image](https://github.com/user-attachments/assets/c394e7ad-3dd1-4096-b275-058d5b55b860)

in detail:
![image](https://github.com/user-attachments/assets/3747e366-d1f4-45de-b7d2-7904525517af)

Maybe the package is missing `pkg:maven...`?

**Comment by aikebah** [comment]
did you disable the CentalAnalyzer?

**Comment by itineric** [comment]
> did you disable the CentalAnalyzer?

I dont think so

**Comment by aikebah** [comment]
If not, your simple-xml-safe.jar would not match any of the published versions of the same library in maven central.

As the CLI cannot establish coordinates it falls back to pure text-matching and the suppressions that we host will always be created for known package-urls. 

Since your jar-file apparently cannot be linked to a package-url via sha1 search on maven central (otherwise you'd have had a pkg-url in the identifiers when the central analyze is indeed not switch off by you) you'd have to resolve this one by your own suppression-file.

**Comment by itineric** [comment]
After some checks: I did not disable central analyzer.

the sha1 on central is here: https://repo1.maven.org/maven2/com/carrotsearch/thirdparty/simple-xml-safe/2.7.1/simple-xml-safe-2.7.1.jar.sha1
it's exactly the same as dependency check shows in his report (screenshot above).

and it can be found using the sha1 search on maven central : https://search.maven.org/solrsearch/select?q=1:045fda5ac6087bc82a209d8cdb73f8d0dbdcfc7b&rows=20&wt=json
also found on alternate central: https://central.sonatype.com/solrsearch/select?q=1:045fda5ac6087bc82a209d8cdb73f8d0dbdcfc7b&rows=20&wt=json

In conclusion: I may have not understood what you are explaining to me but on my side I still having false positive with standard config

**Comment by aikebah** [comment]
Well, if central analyzer has run then, given that there is a sha match with the artifact as hosted on maven central it should've recorded the G/A/V for the library (which would also show as a `pkg:maven/....` identifier in the report), which in turn would be what the hosted suppression will use as an anchor to suppress the simplexml project's CPE.

However the screenshot you shared is not showing a maven identifier for the library, so either CentralAnalyzer was not active on that scan, or it ran into some error in its operation.

**Comment by itineric** [comment]
I managed to reproduce the issue, it is linked to the usage of an alternate "central url".
How to reproduce:

Get the database once (or use some local database if available)
```
> mkdir dependency-check-data

> docker run -it --rm \
    -e user=$USER \
    -u $(id -u ${USER}):$(id -g ${USER}) \
    -v ./dependency-check-data:/usr/share/dependency-check/data \
    owasp/dependency-check:10.0.4 --updateonly --nvdApiKey $NVD_API_KEY

# backup the database to avoid another download
> cp -r dependency-check-data dependency-check-data.bak

```

Get the jar
```
> mkdir src
> cd src
> wget https://repo1.maven.org/maven2/com/carrotsearch/thirdparty/simple-xml-safe/2.7.1/simple-xml-safe-2.7.1.jar
> cd ..
```


Run on default central:
```
> docker run -it --rm \
    -e user=$USER \
    -u $(id -u ${USER}):$(id -g ${USER}) \
    -v ./dependency-check-data:/usr/share/dependency-check/data \
    -v ./src:/src \
    owasp/dependency-check:10.0.4 --noupdate -s /src
```
**False positive is correclty filtered, no CVE found.**

Clean the data (especially to clean the cache)
```
> rm -rf dependency-check-data
> cp -r dependency-check-data.bak dependency-check-data
```

Rerun with the alternate central:
```
> docker run -it --rm \
    -e user=$USER \
    -u $(id -u ${USER}):$(id -g ${USER}) \
    -v ./dependency-check-data:/usr/share/dependency-check/data \
    -v ./src:/src \
    owasp/dependency-check:10.0.4 --noupdate -s /src \
      --centralUrl "https://central.sonatype.com/solrsearch/select"
```
**False positive pops as a critical CVE**.

**Comment by jeremylong** [comment]
thank you for finding the solution - don't use an alternative centralUrl. If you do, create and maintain your own suppression. and hint files as necessary?

**Comment by itineric** [comment]
Do you have some details on how "package" is defined (how does the match with packageUrl from suppression work)? I did not find that explanation on the "Suppressing False Positives" page of the doc. 

And for my knownledge, why are suppressions using "packageUrl" and not sha1 in the default delivered FP file?
